scripts/build-plugins-report.js

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+const open = require('open');
+const { execFile: execFileCb } = require('child_process');
+const { promisify } = require('util');
+
+const execFile = promisify(execFileCb);
+
+const rootDirectory = path.resolve(__dirname, '..');
+
+async function run(command, ...args) {
+  const { stdout, stderr } = await execFile(command, args, {
+    cwd: rootDirectory,
+  });
+
+  if (stderr) {
+    console.error(stderr);
+  }
+
+  return stdout.trim();
+}
+
+function generateHtmlReport(rows) {
+  const thead = `<tr>${rows[0].map(cell => `<th>${cell}</th>`).join('')}</tr>`;
+
+  const tbody = rows
+    .slice(1)
+    .map(row => `<tr>${row.map(cell => `<td>${cell}</td>`).join('')}</tr>`)
+    .join('');
+
+  return `
+    <html lang="en">
+      <head>
+        <title>Backstage Plugins Report</title>
+        <style>
+          * {font-family:sans-serif;}
+          table {width:100%; border-collapse: collapse;}
+          table,td,th {border:1px solid;}
+        </style>
+      </head>
+      <body>
+        <table>
+          <thead>${thead}</thead>
+          <tbody>${tbody}</tbody>
+        </table>
+      </body>
+    </html>
+  `;
+}
+
+const PLUGIN_ROLES = ['frontend-plugin', 'backend-plugin'];
+
+async function main() {
+  const pluginsDirectory = path.resolve(rootDirectory, 'plugins');
+
+  const directoryNames = fs
+    .readdirSync(pluginsDirectory, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name);
+
+  const tableRows = [['Plugin', 'Author', 'Message', 'Date']];
```

Comment:
Would you like me to sort the results by date?

Reply:
Done 🙌🏻 : https://github.com/backstage/backstage/pull/20487/commits/a8aaef7674d6a75095d302c15b75232f2813bc3f